### PR TITLE
add associations

### PR DESCRIPTION
**Pull request by osbentech**
In this project, I worked on 

    - New Classroom, Rental, and Book classes
    - Has-many/belongs-to relationship between Classroom and Student
    - Many-to-many relationship between Person and Book using the intermediate class Rental

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  attr_accessor :title, :author, :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(person, book, date)
+    rental = Rental.new(person, book, date)
```
Comment (elisha2kyakpo1) [inline book.rb:13]: - [ ] [OPTIONAL] For a `book` parameter you can just write `self`

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi 👋 @osbentech, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ The PR title and description look good and professional.
✔️ No linter errors.
✔️ Descriptive readme file.

## Required Changes ♻️
_Check the comments under the review._

- [ ] Kindly fix the License broken link which is giving a 404 error.

license


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by DammyShittu** [comment]
Hi @osbentech,

While you made a great effort in this project, I cannot proceed to review your code.

## Invalid Code Review Request

### Kindly ensure you push your changes before requesting for review. Your MIT link in the readme is still broken as pointed out earlier.

image

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by VICTOR-CHUKWU — CHANGES_REQUESTED** [review]
Hi @osbentech , 
## STATUS: CHANGES REQUIRED :recycle: 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
almost

 ## Required Changes ♻️
_Check the comments under the review._

- [ ] Your MIT link is still broken in your README file, you can add this by creating a file in your root directory called MIT.md, copy paste microverse MIT to the file, then in your README file, change your line 106 to `This project is [MIT](./MIT.md) licensed.`. i hope this makes sense

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by sumairq — APPROVED** [review]
### Hi @osbentech , 



Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✔️ Good commit history
✔️Linters are passing
✔️Very professional README
✔️Correct Git flow
✔️Descriptive PR
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**